### ISSUE TITLE
Support for whois response code 330 (RPL_WHOIS_ACCOUNT)

### ISSUE THREAD

**Issue by eklitzke** [weechat/weechat#1160]
Oragono (an ircd) is using response code 330 in whois queries to return the "authname" of authenticated users (i.e. the name of the account the authed user is associated with). Weechat formats these responses poorly. From what I can tell this is a non-standard IRC response code, so it's understandable that weechat doesn't support it; but it looks simple enough so I'd like to put it forward for consideration. The original bug report I filed in oragono can be found here if you want to take a look: https://github.com/oragono/oragono/issues/217 . Weechat also formats the language response improperly, I'm not sure off hand what code oragono is using.

**Comment by flashcode** [comment]
Hi,
Could you please paste here an example of the response received as you can see it in `/server raw`?
Thanks.

**Comment by eklitzke** [comment]
```
02:10 <-- dorks WHOIS shivaram
02:10 --> dorks :irc.darwin.network 311 evan shivaram ~shivaram c-107-3-81-17.hsd1.ma.comcast.net * shivaram
02:10 --> dorks :irc.darwin.network 319 evan shivaram :#darwin 
02:10 --> dorks :irc.darwin.network 671 evan shivaram :is using a secure connection
02:10 --> dorks :irc.darwin.network 330 evan shivaram :is logged in as
02:10 --> dorks :irc.darwin.network 690 evan shivaram en :can speak these languages
02:10 --> dorks :irc.darwin.network 317 evan shivaram 11749 1521035820 :seconds idle, signon time
02:10 --> dorks :irc.darwin.network 318 evan shivaram :End of /WHOIS list
02:10 <-- dorks PING irc.darwin.network
02:10 --> dorks :irc.darwin.network PONG irc.darwin.network
```

**Comment by flashcode** [comment]
Thanks.

**Comment by eklitzke** [comment]
Here's another weird one, what I see when I `/whois` myself.

Raw server data:
```
02:15 <-- dorks WHOIS evan
02:15 --> dorks :irc.darwin.network 311 evan evan ~death hacker.monad.io * death
02:15 --> dorks :irc.darwin.network 319 evan evan :#darwin 
02:15 --> dorks :irc.darwin.network 338 evan evan ~death@hacker.monad.io 75.101.96.6 :Actual user@host, Actual IP
02:15 --> dorks :irc.darwin.network 671 evan evan :is using a secure connection
02:15 --> dorks :irc.darwin.network 330 evan evan :is logged in as
02:15 --> dorks :irc.darwin.network 690 evan evan en :can speak these languages
02:15 --> dorks :irc.darwin.network 317 evan evan 293 1521159054 :seconds idle, signon time
02:15 --> dorks :irc.darwin.network 318 evan evan :End of /WHOIS list
```

How WeeChat renders it:
```
02:15 -- [evan] (~death@hacker.monad.io): death
02:15 -- [evan] #darwin
02:15 -- [evan] 75.101.96.6 :Actual user@host, Actual IP ~death@hacker.monad.io
02:15 -- [evan] is using a secure connection
02:15 -- [evan] logged in as :is
02:15 -- evan en :can speak these languages
02:15 -- [evan] idle: 00 hours 04 minutes 53 seconds, signon at: Thu, 15 Mar 2018 17:10:54
02:15 -- [evan] End of /WHOIS list
```

In this case the 338 response with my actual IP is rendered kind of strangely.

**Comment by flashcode** [comment]
Hi,

It seems it has been fixed in version 3.4 where the IRC parser has been improved with a lot of changes.

The output in version 4.0.4 is now:

```
-- [evan] is logged in as
```
